src/sass/parse.js

```diff
@@ -4177,34 +4180,68 @@ function getSC() {
 }
 
 /**
- * Check if token is part of a hexadecimal number (e.g. `#fff`) inside
- *      a simple selector
+ * Check if token is part of a hexadecimal number (e.g. `#fff`) inside a simple
+ * selector
  * @param {number} i Token's index number
  * @return {number}
  */
 function checkShash(i) {
-  var l;
+  const start = i;
+  let l;
 
-  if (i >= tokensLength || tokens[i].type !== TokenType.NumberSign) return 0;
+  if (i >= tokensLength) return 0;
+
+  if (tokens[i].type === TokenType.NumberSign) {
+    if (checkInterpolation(i)) return 0;
```

Comment:
Why do we need to check for interpolation here? If that's the case then next character after `#` will be `{` which is an invalid ident and the check will fail.